### ISSUE TITLE
Windows NMAKE Error

### ISSUE THREAD

**Issue by zay448345045** [openssl/openssl#17906]
Hey all, first post here so feel free to let me have it if i'm doing this wrong.

I'm experiencing an issue during build/install when running 'nmake' from my Windows terminal following closely the steps outlined in NOTES-WINDOWS.md and issuing 'perl Configure VC-WIN64A". The perl command appears to execute successfully.

Here's the section that seems to produce the issue. I can run again with a longer buffer if needed:
```
    Microsoft (R) Windows (R) Resource Compiler Version 14.31.31104.0
    Copyright (C) Microsoft Corporation.  All rights reserved.

        "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\nmake.exe" /                   _build_sw

Microsoft (R) Windows (R) Resource Compiler Version 14.31.31104.0
Copyright (C) Microsoft Corporation.  All rights reserved.


        "cl"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MDd /W3 /wd4090 /nologo /Od -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"DEBUG" -D"_DEBUG"   -c /Foapps\lib\libapps-lib-app_libctx.obj "apps\lib\app_libctx.c"
app_libctx.c
C:\Users\zcbmp\Downloads\Compressed\openssl-3.0.2\openssl-3.0.2\include\openssl/crypto.h(25): fatal error C1083: Cannot open include file: “stdlib.h”: No such file or directory
NMAKE : fatal error U1077: “"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\cl.EXE"”: return code“0x2”
Stop.
NMAKE : fatal error U1077: “"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\nmake.exe"”: return code“0x2”
Stop.
```
Thanks in advance for any help on this!

**Comment by mattcaswell** [comment]
> Cannot open include file: “stdlib.h”: No such file or directory

This is one of the standard header files that should be included with your copy of VisualStudio. This suggests some environmental problem (i.e. assuming VisualStudio is correctly installed it should be there - it just can't find it for some reason).

This step from NOTES-WINDOWS.md is key for setting up the environment:

> 4. Use Visual Studio Developer Command Prompt with administrative privileges,
>    choosing one of its variants depending on the intended architecture.
 >   Or run "cmd" and execute "vcvarsall.bat" with one of the options x86,
 >   x86_amd64, x86_arm, x86_arm64, amd64, amd64_x86, amd64_arm, or amd64_arm64.
 >   This sets up the environment variables needed for nmake.exe, cl.exe, etc.
 >   See also
 >   <https://docs.microsoft.com/cpp/build/building-on-the-command-line>

How did you start your Visual Studio Developer Command Prompt?

**Comment by zay448345045** [comment]
> > Cannot open include file: “stdlib.h”: No such file or directory
> 
> This is one of the standard header files that should be included with your copy of VisualStudio. This suggests some environmental problem (i.e. assuming VisualStudio is correctly installed it should be there - it just can't find it for some reason).
> 
> This step from NOTES-WINDOWS.md is key for setting up the environment:
> 
> > 4. Use Visual Studio Developer Command Prompt with administrative privileges,
> >    choosing one of its variants depending on the intended architecture.
> >    Or run "cmd" and execute "vcvarsall.bat" with one of the options x86,
> >    x86_amd64, x86_arm, x86_arm64, amd64, amd64_x86, amd64_arm, or amd64_arm64.
> >    This sets up the environment variables needed for nmake.exe, cl.exe, etc.
> >    See also
> >    https://docs.microsoft.com/cpp/build/building-on-the-command-line
> 
> How did you start your Visual Studio Developer Command Prompt?

Both win system's start menu and of "commands prompt" View Windows in visual studio.Thank you very much.

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input

**Comment by zay448345045** [comment]
Thanks for answer. It's about environmental configuration. Resolved the issue after re adding environment path.And successfully compiled.
